virtool/tasks/runner.py

```diff
@@ -23,20 +23,32 @@ def __init__(
     async def run(self):
         logging.info("Started task runner")
 
+        tasks = []
+
         try:
             while True:
                 logging.info("Waiting for next task")
 
                 task_id = await self._tasks_client.pop()
 
-                await self.run_task(task_id)
+                tasks.append(self.run_task(task_id))
+
+                await asyncio.gather(*tasks)
 
                 logging.info("Finished task: %s", task_id)
 
         except asyncio.CancelledError:
-            logging.info("Stopped task runner")
+            logging.info("Stopped task runner; awaiting task completion")
+
+            await asyncio.wait_for(asyncio.gather(*tasks), timeout=20)
```

Comment:
My bad on this one, but lets up it to 10 mins. Very long vs the default, but some of the tasks are long running and I want to give them a chance to complete before the container getting killed. Easy enough to increase grace period in Kube.